spec/lib/ruy/conditions/in_cyclic_order.rb

```diff
@@ -0,0 +1,108 @@
+require 'spec_helper'
+
+describe Ruy::Conditions::InCyclicOrder do
+  describe '#call' do
+    context 'when cycle is defined in order' do
+      subject(:condition) { Ruy::Conditions::InCyclicOrder.new(0, 17, :age) }
+
+      it 'is true when in range' do
+        context = Ruy::VariableContext.new({:age => 10}, {})
+
+        expect(condition.call(context)).to be
+      end
+
+      it 'is false when out of range' do
+        context = Ruy::VariableContext.new({:age => 18}, {})
+
+        expect(condition.call(context)).to_not be
+      end
+
+      context 'when nested conditions' do
+        subject(:condition) do
+          Ruy::Conditions::InCyclicOrder.new(0, 17, :age) do
+            assert :success
+          end
+        end
+
+        it 'is true when nested succeeds' do
+          context = Ruy::VariableContext.new({:age => 10, :success => true}, {})
+
+          expect(condition.call(context)).to be
+        end
+
+        it 'is false when nested fails' do
+          context = Ruy::VariableContext.new({:age => 10, :success => false}, {})
+
+          expect(condition.call(context)).to_not be
+        end
+      end
+    end
+
+    context 'when cycle is defined in inverse order' do
+      subject(:condition) { Ruy::Conditions::InCyclicOrder.new(23, 5, :time) }
+
+      it 'is true when in cycle' do
+        context = Ruy::VariableContext.new({:time => 1}, {})
+
+        expect(condition.call(context)).to be
+      end
+
+      it 'is false when out of cycle' do
+        context = Ruy::VariableContext.new({:time => 22}, {})
+
+        expect(condition.call(context)).to_not be
+      end
+
+      context 'when nested conditions' do
```

Comment:
Factor this out to a shared example, and include it in both 'when cycle is defined in order' and ''when cycle is defined in inverse order' contexts